avocado/core/job.py

```diff
@@ -221,6 +222,22 @@ def __stop_job_logging(self):
             for logger in loggers:
                 logging.getLogger(logger).removeHandler(handler)
 
+    def _create_test_suites(self):
+        """Creates a list of TestSuite() objects and check for found tests."""
+        test_suites = []
+        for config in self.suites_configs:
+            try:
```

Comment:
I was looking for the replaced block (since the same functionality already exists), and couldn't find.  So, I'd suggest you do something like the following, which would both decrease the complexity of this method and avoid duplication from the start:

```diff
diff --git a/avocado/core/job.py b/avocado/core/job.py
index bcf41365a..f9e57218a 100644
--- a/avocado/core/job.py
+++ b/avocado/core/job.py
@@ -222,19 +222,24 @@ class Job:
             for logger in loggers:
                 logging.getLogger(logger).removeHandler(handler)
 
+    @staticmethod
+    def _create_test_suite(config):
+        try:
+            test_suite = TestSuite.from_config(config)
+            if test_suite.size == 0:
+                refs = test_suite.references
+                msg = ("No tests found for given test references, try "
+                       "'avocado list -V %s' for details") % " ".join(refs)
+                raise exceptions.JobTestSuiteEmptyError(msg)
+        except TestSuiteError as details:
+            raise exceptions.JobBaseException(details)
+        return test_suite
+
     def _create_test_suites(self):
         """Creates a list of TestSuite() objects and check for found tests."""
         test_suites = []
         for config in self.suites_configs:
-            try:
-                test_suite = TestSuite.from_config(config)
-                if test_suite.size == 0:
-                    refs = test_suite.references
-                    msg = ("No tests found for given test references, try "
-                           "'avocado list -V %s' for details") % " ".join(refs)
-                    raise exceptions.JobTestSuiteEmptyError(msg)
-            except TestSuiteError as details:
-                raise exceptions.JobBaseException(details)
+            test_suite = self._create_test_suite(config)
             test_suites.append(test_suite)
         return test_suites
 
@@ -450,15 +455,7 @@ class Job:
                     pass
 
     def create_test_suite(self):
-        try:
-            self.test_suite = TestSuite.from_config(self.config)
-            if self.test_suite.size == 0:
-                refs = self.test_suite.references
-                msg = ("No tests found for given test references, try "
-                       "'avocado list -V %s' for details") % " ".join(refs)
-                raise exceptions.JobTestSuiteEmptyError(msg)
-        except TestSuiteError as details:
-            raise exceptions.JobBaseException(details)
+        self.test_suite = self._create_test_suite(self.config)
         self.result.tests_total = self.test_suite.size
 
     def post_tests(self):
```

Reply:
Yes, I will do this. makes sense the refactoring.